includes/blocks.inc

```diff
@@ -74,23 +74,88 @@ function islandora_basic_collection_clear_object_count_cache() {
  * Outputs a JSON string to the buffer.
  */
 function islandora_basic_collection_object_count_callback() {
+  $count_method = variable_get('islandora_basic_collection_count_method', 'SPARQL');
+
+
   $cid = islandora_basic_collection_get_object_count_block_cache_id();
 
   if ($value = cache_get($cid, 'cache')) {
     $substitutions = $value->data;
   }
   else {
+
     $tuque = islandora_get_tuque_connection();
     $objects_query_array = islandora_basic_collection_get_query_info(array(
       'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
       'collection_listing' => TRUE,
       'all_objects' => TRUE,
     ));
     $collection_query_array = islandora_basic_collection_get_query_info(array(
-      'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
-      'collection_listing' => TRUE,
-      'all_objects' => FALSE,
-    ));
+    'object' => islandora_object_load(variable_get('islandora_repository_pid', 'islandora:root')),
+    'collection_listing' => TRUE,
+    'all_objects' => FALSE,
+  ));
+
+
+  // Setup Solr query for collection and object counts.
+  if ($count_method == "Solr") {
+    $all_cmodels = islandora_get_content_models();
+    $models_to_exclude = variable_get('islandora_basic_collection_object_count_listing_content_models_to_restrict', FALSE);
+    $cmodels_to_count = array();
+    foreach($all_cmodels AS $cmodel) {
+      if ($models_to_exclude[$cmodel['pid']] != '0') {
+        $cmodels_to_count[] = $cmodel['pid'];
+      }
+    }
+
+    // Collection count Solr query.
+    $collection_query = "RELS_EXT_hasModel_uri_mt:collection";
```

Comment:
This is a tokenized field: ours contains RELS_EXT_hasModel_uri_mt:"islandora:collectioncmodel" and it of course It will match collection (because of tokenizing) but the query may be faster if you go for the _ms and match the full CMODEL.

Reply:
Thanks, good call.